File: an/Mingrisoft/04/05/Radio Button/src/main/java/com/mingrisoft/MainActivity.java
package com.mingrisoft;

import android.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    Button bt;                                              //定义提交按钮
    RadioGroup rg;                                         //定义单选按钮组

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        bt = (Button) findViewById(R.id.bt);                //通过ID获取布局提交按钮
        rg = (RadioGroup) findViewById(R.id.rg);           //通过ID获取布局单选按钮组
        bt.setOnClickListener(new View.OnClickListener() {  //为提交按钮设置单击事件监听器
            @Override
            public void onClick(View v) {
                for (int i = 0; i < rg.getChildCount(); i++) {
                    RadioButton radioButton = (RadioButton) rg.getChildAt(i);  //根据索引值获取单选按钮
                    if (radioButton.isChecked()) {                           //判断单选按钮是否被选中
                        if (radioButton.getText().equals("B:100")) {           //判断答案是否正确
                            Toast.makeText(MainActivity.this,
                                    "回答正确", Toast.LENGTH_LONG).show();
                        } else {
                            //错误消息提示框
                            AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
                            builder.setMessage("回答错误，下面请看解析：当张山换完零钱之后，" +
                                    "给了顾客75还有价值25元的商品，自己还剩下了25元。这时，" +
                                    "李石来找张山要钱，张山把自己剩下的相当于是李石的25元给了李石，" +
                                    "另外自己掏了75元。这样张山赔了一个25元的商品和75元的人民币，" +
                                    "总共价值100元。");
                            builder.setPositiveButton("确定", null).show();    //单击确定消失
                        }
                        break;                                           //跳出for循环
                    }
                }
            }
        });
    }
}

File: an/Dictionary/app/src/main/java/com/example/dictionary/MainActivity.kt
package com.example.dictionary

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.ImageButton
import android.widget.ListView
import android.widget.SimpleAdapter
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    private lateinit var dbOpenHelper : DBOpenHelper

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        dbOpenHelper = DBOpenHelper(this)
        val listView = findViewById<ListView>(R.id.result_listView)
        val etSearch = findViewById<EditText>(R.id.search_et)
        val btnSearch = findViewById<Button>(R.id.search_btn)
        val btnAdd = findViewById<Button>(R.id.btn_add)


        btnAdd.setOnClickListener(View.OnClickListener {
            val intent = Intent(this, AddActivity::class.java)
            startActivity(intent)
        })


        btnSearch.setOnClickListener(View.OnClickListener {
            val key = etSearch.text.toString()
            val cursor = dbOpenHelper.readableDatabase.query("dict",
                null, "word = ?", arrayOf<String>(key), null, null, null)
            val resultList = ArrayList<Map<String, String>>()
            while (cursor.moveToNext()) {
                val map = HashMap<String, String>()
                map.put("word", cursor.getString(1))
                map.put("explain", cursor.getString(2))
                resultList.add(map)
            }

            if (!etSearch.isFocused) {
                if(resultList == null || resultList.size == 0) {
                    Toast.makeText(this, "No such word found!", Toast.LENGTH_LONG).show()
                } else {
                    val simpleAdapter = SimpleAdapter(this, resultList, R.layout.result_main,
                        arrayOf("word","explain"), intArrayOf(R.id.result_word, R.id.result_interpret))
                    listView.adapter = simpleAdapter
                }
            }
        })
    }

    override fun onDestroy() {
        super.onDestroy()
        dbOpenHelper.close()
    }
}
